actors/init/src/main/java/org/sunbird/init/actors/SystemInitActor.java

```diff
@@ -0,0 +1,274 @@
+ *
+ * @author Loganathan Shanmugam
+ */
+@ActorConfig(
+  tasks = {"systemInitRootOrg"},
+  asyncTasks = {}
+)
+public class SystemInitActor extends BaseActor {
+  private final CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private ObjectMapper mapper = new ObjectMapper();
+  private SystemSetting systemSetting;
+  private SystemSettingService systemSettingService = new SystemSettingServiceImpl();
+  private Integer writeSettingsRetry = 0;
+  private Integer writeSettingsRetriesAllowed = 1;
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    Util.initializeContext(request, TelemetryEnvKey.ORGANISATION);
+    ExecutionContext.setRequestId(request.getRequestId());
+    if (request.getOperation().equalsIgnoreCase(ActorOperations.SYSTEM_INIT_ROOT_ORG.getValue())) {
+      systemInitRootOrg(request);
+    } else {
+      onReceiveUnsupportedOperation(request.getOperation());
+    }
+  }
+  /**
+   * This Method to initalises the first root org creation after validating the data
+   *
+   * @param actorMessage Instance of Request class contains the organisation data to be created
+   */
+  @SuppressWarnings("unchecked")
+  private void systemInitRootOrg(Request actorMessage) {
+    ProjectLogger.log("systemInitRootOrg method call started");
+    SystemSetting initStatus = getInitialisationStatus();
+    if (isNotNull(initStatus) && isNotNull(initStatus.getValue())) {
+      if (initStatus.getValue().equals(JsonKey.COMPLETED)) {
+        ProjectCommonException.throwClientErrorException(
+            ResponseCode.systemAlreadyInitialised,
+            ResponseCode.systemAlreadyInitialised.getErrorMessage());
+      } else if (initStatus.getValue().equals(JsonKey.STARTED)) {
+        checkAndCreateRootOrg(actorMessage);
+        setInitialisationStatus(JsonKey.COMPLETED);
+      }
+    } else {
+      setInitialisationStatus(JsonKey.STARTED);
+      checkAndCreateRootOrg(actorMessage);
+      setInitialisationStatus(JsonKey.COMPLETED);
+    }
+  }
+/**
+ * This method checks for rootOrg exists or not
+ * 
+ * @return Boolean true if exists else false
+ */
+  private boolean isRootOrgExists() {
+    Map<String, Object> filters = new HashMap<>();
+    filters.put(JsonKey.IS_ROOT_ORG, true);
+    Map<String, Object> esResult =
+        elasticSearchComplexSearch(
+            filters, EsIndex.sunbird.getIndexName(), EsType.organisation.getTypeName());
+    if (isNotNull(esResult)
+        && esResult.containsKey(JsonKey.CONTENT)
+        && isNotNull(esResult.get(JsonKey.CONTENT))
+        && ((List) esResult.get(JsonKey.CONTENT)).size() > 0) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+ * This method creates rootOrg if it is not exists already
+ * 
+ * @param actorMessage instance of Request class with rootOrg data.
+ */
+  private void checkAndCreateRootOrg(Request actorMessage) {
+    if (isRootOrgExists() == false) {
+      Map<String, Object> req =
```

Comment:
Lines 114-126 => factor this into a separate function `getFirstRootOrgParams`

exclude line 117